kolibri2zim/scraper.py

```diff
@@ -1059,13 +1034,58 @@ def add_custom_about_and_css(self):
             if not user_provided_file:
                 content = ""
             else:
-                with open(user_provided_file) as fh:
-                    content = fh.read()
+                content = user_provided_file.read_bytes()
         # otherwise, create a blank one
         else:
             content = ""
 
-        self.creator.add_item_for(
-            "custom.css", content=content.encode(), mimetype="text/css"
-        )
+        self.creator.add_item_for("custom.css", content=content, mimetype="text/css")
         logger.debug("Added about page and custom CSS")
+
+    def ensure_js_deps_are_present(self):
+        for js_deps_file in [
+            "epub.min.js",
+            "jszip.min.js",
+            "jquery.min.js",
+            "videojs-ogvjs.js",
+        ]:
+            if not os.path.exists(
+                self.templates_dir.joinpath(f"assets/{js_deps_file}")
+            ):
+                raise ValueError(
+                    "It looks like JS deps have not been installed,"
+                    f" {js_deps_file} is missing"
+                )
+
+        for js_deps_dir in [
+            "pdfjs",
+            "videojs",
+            "ogvjs",
+            "bootstrap",
+            "bootstrap-icons",
+            "perseus",
+        ]:
+            if not os.path.exists(self.templates_dir.joinpath(f"assets/{js_deps_dir}")):
+                raise ValueError(
+                    "It looks like JS deps have not been installed,"
+                    f" {js_deps_dir} is missing"
+                )
+
+        if self.s3_url_with_credentials and not self.s3_credentials_ok():
```

Comment:
I don't like the rest of this being in a method named `ensure_js_deps_are_present`. Move it back to scraper or to a dedicated method